griptape/drivers/prompt/base_prompt_driver.py

```diff
@@ -76,7 +79,10 @@ def run(self, prompt_stack: PromptStack) -> TextArtifact:
                     tokens = []
                     completion_chunks = self.try_stream(prompt_stack)
                     for chunk in completion_chunks:
-                        self.structure.publish_event(CompletionChunkEvent(token=chunk.value))
+                        if self.structure:
+                            self.structure.publish_event(CompletionChunkEvent(token=chunk.value))
+                        else:
+                            raise ValueError("Streaming requires a structure")
```

Comment:
I agree, but in the current implementation, publishing events requires a Structure to be set. 

I can revert this change since it's kind of out of scope of an already massive PR, but I think we need a separate issue to decouple Prompt Drivers and Structures. Maybe Prompt Drivers take an Event Listener instead.

Reply:
Sounds good. Let's address it in the separate issue.